### ISSUE TITLE
[fs-watch] function watch() callback interface `WatchEvent` Type mismatch. 

### ISSUE THREAD

**Issue by cheeselemon** [tauri-apps/plugins-workspace#2221]
### The Problem: 

The [docs](https://v2.tauri.app/reference/javascript/fs/#watch) state that the watch() function of plugin-fs as follows: 

```
function watch(
   paths,
   cb,
options?): Promise<UnwatchFn>
```

and the `cb` is a function defined as: `(event: WatchEvent) => void`

the interface `WatchEvent` is declared as follows:

```
interface WatchEvent {
  type: WatchEventKind
  paths: string[]
  attrs: unknown
}
``` 

And checking out interface `WatchEventKind`: https://github.com/tauri-apps/plugins-workspace/blob/da64d9b665694f541b1ee2d533535210e118969f/plugins/fs/guest-js/index.ts#L1177 

However, upon implementing the watch() callback function, I found out that the actual data didn't contain `type` instead it had `kind`. 

ie. the interface states that `type : WatchEventKind`, but the data was actually `kind: WatchEventKind` 

Actual callback data sample: 

```
{kind: {modify: {kind: "data", mode: "content"}}, paths: [".../some_path/..."], attrs: {}}
```

And upon checking out the [watcher.rs](https://github.com/tauri-apps/plugins-workspace/blob/v2/plugins/fs/src/watcher.rs) native rust code, I can see that it's also declared as `kind` 

Am I missing something? 

### How to fix? 

As a workaround, I've declared a custom interface for this process in my own source code. 

I think it'd be best if the type declaration was fixed on javascript. 

### Additional Info & Environment 

my Cargo.toml:

```
[dependencies]
...
tauri-plugin-fs = { version = "2.0.0", features = ["watch"] }
...
```

Using verion `2.2.0` for `tauri-plugin-fs` didn't fix. 

Operating system & App Platform : MacOS Silicon

**Comment by Legend-Master** [comment]
Seems like notify changed the serialization in https://github.com/notify-rs/notify/pull/558 and we thought with `serialization-compat-6` feature introduced in https://github.com/notify-rs/notify/pull/568, it would be the same as before but looks like there's a bug so... Will take a look and see if I could open a fix there

**Comment by dfaust** [comment]
I just published `notify-types-1.0.1`, so a `cargo update` should fix the issue.
Thanks again for the PR @Legend-Master.

**Comment by cheeselemon** [comment]
Awesome, thanks a lot ! 💪
